Geometry/TrackerGeometryBuilder/interface/PixelTopologyMap.h

```diff
@@ -0,0 +1,57 @@
+#ifndef Geometry_TrackerGeometryBuilder_PixelTopologyMap_H
+#define Geometry_TrackerGeometryBuilder_PixelTopologyMap_H
+
+// system include files
+#include <map>
+#include <memory>
+#include <iostream>
+#include <iomanip>  // std::setw
+
+// user include files
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "TrackerGeometry.h"
+
+/**
+ * A specific Pixel Tracker class to determine the number of ladders / modules in PXB
+   and number of blades/modules in PXF
+ */
+class PixelTopologyMap {
+public:
+  PixelTopologyMap(const TrackerGeometry* geom, const TrackerTopology* topo)
+      : m_trackerTopo{topo}, m_trackerGeom{geom} {
+    // build the maps
+    buildTopologyMaps();
+  }
+
+  ~PixelTopologyMap() = default;
+
+  // getter methods
+
+  inline const unsigned getPXBLadders(unsigned int lay) const { return m_pxbMap.at(lay).first; }
+  inline const unsigned getPXBModules(unsigned int lay) const { return m_pxbMap.at(lay).second; }
+  inline const unsigned getPXFBlades(int disk) const { return m_pxfMap.at(std::abs(disk)).first; }
+  inline const unsigned getPXFModules(int disk) const { return m_pxfMap.at(std::abs(disk)).second; }
+
+  // printout
+  void printAll(std::ostream& os) const;
+
+private:
+  void buildTopologyMaps();
+
+  const TrackerTopology* m_trackerTopo;
+  const TrackerGeometry* m_trackerGeom;
+
+  std::map<unsigned, std::pair<unsigned, unsigned>> m_pxbMap;
+  std::map<unsigned, std::pair<unsigned, unsigned>> m_pxfMap;
+};
+
+std::ostream& operator<<(std::ostream& os, PixelTopologyMap map) {
```

Comment:
This should have been declared as `inline`, now it leads to one definition rule violations (that broke the C++ modules build, FYI @davidlange6)

Reply:
how does one catch this sort of things?

Reply:
@makortel would this be enough to fix the issue?
https://github.com/cms-sw/cmssw/compare/master...mmusich:fix_PixelTopologyMap?expand=1
I'll make the PR in case (but I don't know how to test it...)

Reply:
> would this be enough to fix the issue?
> https://github.com/cms-sw/cmssw/compare/master...mmusich:fix_PixelTopologyMap?expand=1

That should fix the issue, thanks!

Reply:
https://github.com/cms-sw/cmssw/pull/33363

Reply:
> how does one catch this sort of things?

Currently by eye (essentially all free functions defined in headers should be declared as `inline`, for class member functions it is not required but doesn't hurt). I did find that clang-tidy seems to have [a check for those](https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/misc-definitions-in-headers.html), so we could look into that for autoamted check.

Reply:
> I did find that clang-tidy seems to have [a check for those](https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/misc-definitions-in-headers.html), so we could look into that for autoamted check.

Done in #33365